From cc57af4df4c059380ec69476cb1ffda66dc5af0a Mon Sep 17 00:00:00 2001
From: Shreenidhi Shedi <sshedi@vmware.com>
Date: Mon, 4 Apr 2022 12:59:02 +0530
Subject: [PATCH] Fix AVC denials based on package test results (rebased for 43.6)

Merged with upstream init/systemd changes; omit redundant unconfined mounton
(self:{dir file} mounton already present).

Signed-off-by: Shreenidhi Shedi <sshedi@vmware.com>
---
 policy/modules/admin/usermanage.te     |  1 +
 policy/modules/services/ssh.te         |  1 +
 policy/modules/system/authlogin.te     |  2 ++
 policy/modules/system/init.te          | 32 ++++++++++++++++++++----
 policy/modules/system/iptables.te      |  2 ++
 policy/modules/system/logging.te       |  2 ++
 policy/modules/system/systemd.te       | 14 +++++++++++
 7 files changed, 48 insertions(+), 6 deletions(-)

diff --git a/policy/modules/admin/usermanage.te b/policy/modules/admin/usermanage.te
--- a/policy/modules/admin/usermanage.te
+++ b/policy/modules/admin/usermanage.te
@@ -328,6 +328,7 @@
 allow passwd_t self:msgq create_msgq_perms;
 allow passwd_t self:msg { send receive };
 allow passwd_t self:netlink_selinux_socket create_socket_perms;
+allow passwd_t user_home_t:file write;
 
 allow passwd_t crack_db_t:dir list_dir_perms;
 read_files_pattern(passwd_t, crack_db_t, crack_db_t)

diff --git a/policy/modules/services/ssh.te b/policy/modules/services/ssh.te
--- a/policy/modules/services/ssh.te
+++ b/policy/modules/services/ssh.te
@@ -709,6 +711,7 @@
 
 allow ssh_keygen_t sshd_key_t:file manage_file_perms;
 files_etc_filetrans(ssh_keygen_t, sshd_key_t, file)
+allow ssh_keygen_t etc_t:file unlink;
 
 manage_dirs_pattern(ssh_keygen_t, ssh_home_t, ssh_home_t)
 manage_files_pattern(ssh_keygen_t, ssh_home_t, ssh_home_t)

diff --git a/policy/modules/system/authlogin.te b/policy/modules/system/authlogin.te
--- a/policy/modules/system/authlogin.te
+++ b/policy/modules/system/authlogin.te
@@ -46,6 +46,8 @@
 application_domain(chkpwd_t, chkpwd_exec_t)
 role system_r types chkpwd_t;
 
+libs_use_ld_so(chkpwd_t)
+
 type faillog_t;
 logging_log_file(faillog_t)
 mls_trusted_object(faillog_t)

diff --git a/policy/modules/system/init.te b/policy/modules/system/init.te
--- a/policy/modules/system/init.te
+++ b/policy/modules/system/init.te
@@ -212,7 +214,8 @@
 manage_lnk_files_pattern(init_t, init_tmp_t, init_tmp_t)
 manage_sock_files_pattern(init_t, init_tmp_t, init_tmp_t)
 files_tmp_filetrans(init_t, init_tmp_t, { file sock_file })
-allow init_t init_tmp_t:file map;
+allow init_t init_tmp_t:file { map execute };
+allow init_t self:process execmem;
 
 manage_dirs_pattern(init_t, init_var_lib_t, init_var_lib_t)
 manage_files_pattern(init_t, init_var_lib_t, init_var_lib_t)
@@ -273,6 +276,9 @@
 corenet_all_recvfrom_netlabel(init_t)
 corenet_tcp_bind_all_ports(init_t)
 corenet_udp_bind_all_ports(init_t)
+corenet_udp_bind_generic_node(init_t)
+corenet_tcp_bind_generic_node(init_t)
+corenet_tcp_connect_all_ports(init_t)
 
 dev_create_all_files(init_t)
 dev_create_all_chr_files(init_t)
@@ -337,6 +343,13 @@
 files_watch_non_security_dirs(init_t)
 files_watch_non_security_files(init_t)
 files_watch_non_security_lnk_files(init_t)
+files_manage_etc_files(init_t)
+files_manage_var_dirs(init_t)
+files_manage_var_files(init_t)
+files_manage_generic_spool_dirs(init_t)
+files_manage_generic_spool(init_t)
+files_root_filetrans(init_t, root_t, file)
+files_associate_rootfs(init_t)
 
 fs_cgroup_filetrans_memory_pressure(init_t)
 fs_mounton_cgroup(init_t)
@@ -394,8 +407,6 @@
 init_domtrans_script(init_t)
 init_exec_notrans_direct_init_entry(init_t)
 
-libs_rw_ld_so_cache(init_t)
-
 logging_create_devlog_dev(init_t)
 logging_create_syslog_pid_file(init_t)
 logging_send_syslog_msg(init_t)
@@ -435,6 +446,10 @@
 userdom_watch_user_ptys(init_t)
 userdom_watch_reads_user_ptys(init_t)
 
+userdom_manage_tmp_files(init_t)
+userdom_manage_user_tmp_dirs(init_t)
+userdom_map_tmp_files(init_t)
+
 allow init_t self:process setsched;
 
 ifdef(`distro_gentoo',`
@@ -1198,6 +1213,7 @@
 libs_rw_ld_so_cache(initrc_t)
 libs_exec_lib_files(initrc_t)
 libs_exec_ld_so(initrc_t)
+libs_use_ld_so(init_t)
 
 logging_send_audit_msgs(initrc_t)
 logging_send_syslog_msg(initrc_t)
@@ -2080,3 +2096,7 @@
         ccs_read_config(daemon)
     ')
  ')
+
+optional_policy(`
+    rabbitmq_admin(init_t, system_r)
+')

diff --git a/policy/modules/system/iptables.te b/policy/modules/system/iptables.te
--- a/policy/modules/system/iptables.te
+++ b/policy/modules/system/iptables.te
@@ -13,6 +13,8 @@
 init_system_domain(iptables_t, iptables_exec_t)
 role iptables_roles types iptables_t;
 
+libs_read_lib_files(iptables_t)
+
 type iptables_initrc_exec_t;
 init_script_file(iptables_initrc_exec_t)
 

diff --git a/policy/modules/system/logging.te b/policy/modules/system/logging.te
--- a/policy/modules/system/logging.te
+++ b/policy/modules/system/logging.te
@@ -118,6 +118,8 @@
 init_nnp_daemon_domain(syslogd_t)
 mls_trusted_object(syslogd_t)
 
+libs_read_lib_files(syslogd_t)
+
 type syslogd_initrc_exec_t;
 init_script_file(syslogd_initrc_exec_t)
 

diff --git a/policy/modules/system/systemd.te b/policy/modules/system/systemd.te
--- a/policy/modules/system/systemd.te
+++ b/policy/modules/system/systemd.te
@@ -350,6 +350,8 @@
 allow systemd_logind_t self:netlink_kobject_uevent_socket create_socket_perms;
 allow systemd_logind_t self:unix_dgram_socket create_socket_perms;
 
+libs_read_lib_files(systemd_logind_t)
+
 kernel_rw_psi(systemd_logind_t)
 
 mls_file_read_all_levels(systemd_logind_t)
@@ -678,6 +680,8 @@
 allow systemd_networkd_t self:rawip_socket create_socket_perms;
 allow systemd_networkd_t self:tun_socket { relabelfrom relabelto create_socket_perms };
 
+libs_read_lib_files(systemd_networkd_t)
+
 allow init_t systemd_networkd_t:netlink_route_socket create_netlink_socket_perms;
 
 allow systemd_networkd_t systemd_networkd_var_lib_t:dir {create_dir_perms list_dir_perms};
@@ -907,7 +911,7 @@
 ')
 
 optional_policy(`
-	# we have /run/user/$USER/dconf 
+	# we have /run/user/$USER/dconf
 	gnome_delete_home_config(systemd_tmpfiles_t)
 	gnome_delete_home_config_dirs(systemd_tmpfiles_t)
 	gnome_setattr_home_config_dirs(systemd_tmpfiles_t)
@@ -1055,6 +1059,9 @@
 allow systemd_hostnamed_t self:unix_stream_socket create_stream_socket_perms;
 allow systemd_hostnamed_t self:unix_dgram_socket create_socket_perms;
 
+manage_dirs_pattern(systemd_hostnamed_t, init_var_run_t, init_var_run_t)
+manage_files_pattern(systemd_hostnamed_t, init_var_run_t, init_var_run_t)
+
 manage_files_pattern(systemd_hostnamed_t, hostname_etc_t, hostname_etc_t)
 manage_lnk_files_pattern(systemd_hostnamed_t, hostname_etc_t, hostname_etc_t)
 files_etc_filetrans(systemd_hostnamed_t, hostname_etc_t, file)
@@ -1317,6 +1324,8 @@
 kernel_read_security_state(systemd_sysctl_t)
 kernel_write_security_state(systemd_sysctl_t)
 
+libs_read_lib_files(systemd_sysctl_t)
+
 files_read_system_conf_files(systemd_sysctl_t)
 
 # fs.binfmt_misc.status
@@ -1518,6 +1527,8 @@
 dontaudit systemd_gpt_generator_t self:capability sys_admin;
 allow systemd_gpt_generator_t self:netlink_kobject_uevent_socket create_socket_perms;
 
+libs_read_lib_files(systemd_gpt_generator_t)
+
 dev_read_rand(systemd_gpt_generator_t)
 
 files_list_boot(systemd_gpt_generator_t)
@@ -1779,6 +1790,8 @@
 
 allow systemd_modules_load_t self:system module_load;
 
+libs_read_lib_files(systemd_modules_load_t)
+
 kernel_dgram_send(systemd_modules_load_t)
 kernel_load_unsigned_module(systemd_modules_load_t)
 kernel_ib_access_unlabeled_pkeys(systemd_modules_load_t)

